msm/src/precomputed_srs.rs

```diff
@@ -1,18 +1,116 @@
 //! Clone of kimchi/precomputed_srs.rs but for MSM project with BN254
 
+use crate::{Fp, BN254, DOMAIN_SIZE};
 use ark_ff::UniformRand;
-
+use ark_serialize::Write;
 use kimchi::circuits::domains::EvaluationDomains;
 use poly_commitment::pairing_proof::PairingSRS;
-
-use crate::{Fp, BN254};
+use rand::{prelude::StdRng, SeedableRng};
+use std::{fs::File, io::BufReader, path::PathBuf};
 
 /// Obtains an SRS for a specific curve from disk, or generates it if absent.
 pub fn get_bn254_srs(domain: EvaluationDomains<Fp>) -> PairingSRS<BN254> {
```

Comment:
As discussed in our call some minutes ago, I am in favor of having the deserializer as you implemented, but not change the actual serializer used currently. I am sure there should be a way to use a custom serializer with serde.